components/brave_referrals/browser/brave_referrals_service.cc

```diff
@@ -95,10 +108,18 @@ void BraveReferralsService::Start() {
   if (initialized_)
     return;
 
-  // Retrieve first run sentinel creation time.
-  task_runner_->PostTask(FROM_HERE,
-                         base::Bind(&BraveReferralsService::GetFirstRunTime,
-                                    base::Unretained(this)));
+  // Retrieve first run time.
+  GetFirstRunTime();
+
+  // Periodically perform finalization checks.
+  DCHECK(!finalization_checks_timer_);
+  finalization_checks_timer_ = std::make_unique<base::RepeatingTimer>();
+  finalization_checks_timer_->Start(
+      FROM_HERE,
+      base::TimeDelta::FromSeconds(
+          brave_base::random::Geometric(kFinalizationChecksFrequency)),
+      this, &BraveReferralsService::OnFinalizationChecksTimerFired);
+  DCHECK(finalization_checks_timer_->IsRunning());
```

Comment:
We should run the finalization checks on a timer, not just at startup. Not relevant to this PR, but seemed worthwhile to fix while I was in here.